django/db/models/fields/related_descriptors.py

```diff
@@ -311,7 +317,7 @@ def RelatedObjectDoesNotExist(self):
         )
 
     def is_cached(self, instance):
-        return hasattr(instance, self.cache_name)
+        return self.related.is_cached(instance)
```

Comment:
This was the source for a lot of confusion for me...

Given the example:

```python
class Restaurant(Model):
            place = OneToOneField(Place, related_name='restaurant')
```

`self.related` is the reverse restaurant field on the `Place` model (`place.restaurant`)
`self.related.field` is the place field on the `Restaurant` model.

So `is_cached()` should check if the reverse `restaurant` field is cached on `Place`